### PR TITLE
New: Added pdf printing from scan result (browser extension)

### LINKED ISSUE #3943: New Feature | print/save report generated in browser extension
Hi Web Hint Team,

I have just started using Webhint.io browser extension with both Firefox and Chrome and find it awesome. There is one thing which i think is absolutely required and that's a way to print (to pdf) / save the results generated.

I hope you will find this request genuine.

Regards,
Abhishek Jain

### PR DESCRIPTION
**Pull request by ruppysuppy**
## Pull request checklist

Make sure you:

- [X] Signed the [Contributor License Agreement](https://cla.js.foundation/webhintio/hint)
- [X] Followed the [commit message guidelines](https://webhint.io/docs/contributor-guide/getting-started/pull-requests/#commit-messages)

For non-trivial changes, please make sure you also:

- [ ] Added/Updated related documentation.
- [ ] Added/Updated related tests.

## Short description of the change(s)

Added scan result PDF generation in Browser Extension

-----------------------------------------------------------------------------------------------------------------------

Fix #3943

### CODE REVIEW

**Comment by sarvaje** [comment]
@ruppysuppy thank you for submit the PR!!

Can you add an screenshot?

File: packages/extension-browser/src/devtools/views/controls/print-link.tsx
Comment (sarvaje) [inline packages/extension-browser/src/devtools/views/controls/print-link.tsx:12]: This needs to be localized.
To do so, you need to add a new entry to the file `src/locales/en/messages.json` (e.g. `print`)
and then use the method `getMessage` (e.g. getMessage('print')).

File: packages/extension-browser/src/devtools/views/controls/print-link.tsx
Comment (sarvaje) [inline packages/extension-browser/src/devtools/views/controls/print-link.tsx:11]: I think we should use a `<button>` instead of an span.

**Review by sarvaje — CHANGES_REQUESTED** [review]
Looking good, just a couple of little things.

File: packages/extension-browser/src/devtools/views/controls/print-link.tsx
Comment (ruppysuppy) [reply packages/extension-browser/src/devtools/views/controls/print-link.tsx:12]: I didn't get the getMessage part, where do I need to add it? I'll add the part in the message.json

Comment (sarvaje) [reply packages/extension-browser/src/devtools/views/controls/print-link.tsx:12]: We have an i18n util in `src/devtools/utils/i18n.ts` that export a method `getMessage`, this method will take of the localization. The idea is replace you hardcoded `Print` for a `{getMessage('print')}`

You need to add a new entry to the file `src/_locales/en/messages.json` with you new string e.g.
```json
"print": {
    "message": "Print",
    "description": "Label for button to print the result."
}
```

You can find an example on how we use it in the file `src/devtools/views/pages/results/header.tsx`.

**Comment by ruppysuppy** [comment]
Thanks a lot @sarvaje

**Comment by ruppysuppy** [comment]
__Screenshots:__
_Browser Extension:_



_Generated PDF:_

**Review by sarvaje — APPROVED** [review]

**Comment by sarvaje** [comment]
Looks good!!

File: packages/extension-browser/src/devtools/views/controls/print-link.css
Comment (antross) [inline packages/extension-browser/src/devtools/views/controls/print-link.css:5]: Pretty minor, but some platforms have some rounding on their buttons we'll need to disable here to avoid a slight distortion of the underline:
image


```suggestion
    border-bottom: var(--border);
    border-radius: 0;
```

**Review by antross — CHANGES_REQUESTED** [review]
One minor CSS tweak I think we should make before merging, but everything else looks great! Thanks @ruppysuppy!

**Comment by antross** [comment]
@ruppysuppy FYI don't worry about the Windows tests timing out. We've had some periodic trouble with those. As long as we're not getting any failures and both Mac+Linux pass this will be good to go once we make that final CSS fix.

**Review by antross — APPROVED** [review]
Looks good. Thanks for the PR and the updates @ruppysuppy!

**Pull request merged**
